src/libtiled/tilelayer.h

```diff
@@ -212,6 +212,160 @@ inline const Cell &Chunk::cellAt(const QPoint &point) const
 class TILEDSHARED_EXPORT TileLayer : public Layer
 {
 public:
+    class iterator
+    {
+    public:
+        iterator(QHash<QPoint, Chunk>::iterator it, QHash<QPoint, Chunk>::iterator end)
+            : mChunkPointer(it)
+            , mChunkEndPointer(end)
+        {
+            if (it != end)
+                mCellPointer = it.value().begin();
+        }
+
+        iterator operator++(int)
+        {
+            iterator it = *this;
+
+            if (mChunkPointer != mChunkEndPointer) {
+                if (++mCellPointer == mChunkPointer.value().end()) {
+                    mChunkPointer++;
+                    if (mChunkPointer != mChunkEndPointer)
+                        mCellPointer = mChunkPointer.value().begin();
+                }
+            }
+
+            return it;
+        }
+
+        iterator &operator++()
+        {
+            if (mChunkPointer != mChunkEndPointer) {
+                if (++mCellPointer == mChunkPointer.value().end()) {
+                    mChunkPointer++;
+                    if (mChunkPointer != mChunkEndPointer)
+                        mCellPointer = mChunkPointer.value().begin();
+                }
+            }
+
+            return *this;
+        }
+
+        Cell &operator*() { return *mCellPointer; }
+
+        QVector<Cell>::iterator operator->() { return mCellPointer; }
+
+        friend bool operator==(const iterator& lhs, const iterator& rhs)
+        {
+            if (lhs.mChunkPointer == lhs.mChunkEndPointer || rhs.mChunkPointer == rhs.mChunkEndPointer)
+                return lhs.mChunkPointer == rhs.mChunkPointer;
+            else
+                return lhs.mCellPointer == rhs.mCellPointer;
+        }
+
+        friend bool operator!=(const iterator& lhs, const iterator& rhs)
+        {
+            if (lhs.mChunkPointer == lhs.mChunkEndPointer || rhs.mChunkPointer == rhs.mChunkEndPointer)
+                return lhs.mChunkPointer != rhs.mChunkPointer;
+            else
+                return lhs.mCellPointer != rhs.mCellPointer;
+        }
+
+        Cell &value() { return *mCellPointer; }
```

Comment:
This this operator doesn't change the iterator, it should still be marked `const`.